src/rvoice/fluid_rvoice_mixer.c

```diff
@@ -668,18 +702,25 @@ new_fluid_rvoice_mixer(int buf_count, int fx_buf_count, fluid_real_t sample_rate
 
     FLUID_MEMSET(mixer, 0, sizeof(fluid_rvoice_mixer_t));
     mixer->eventhandler = evthandler;
+    mixer->fx_units = fx_units;
     mixer->buffers.buf_count = buf_count;
-    mixer->buffers.fx_buf_count = fx_buf_count;
+    mixer->buffers.fx_buf_count = fx_buf_count * fx_units;
 
     /* allocate the reverb module */
-    mixer->fx.reverb = new_fluid_revmodel(sample_rate);
-    mixer->fx.chorus = new_fluid_chorus(sample_rate);
-
-    if(mixer->fx.reverb == NULL || mixer->fx.chorus == NULL)
+    mixer->fx = FLUID_ARRAY(fluid_mixer_fx_t, fx_units);
```

Comment:
May be the returned pointer this should be checked ?

Reply:
Thanks!